superset-frontend/src/components/ListView/TableCollection.tsx

```diff
@@ -260,6 +264,7 @@ export default function TableCollection({
             prepareRow(row);
             return (
               <tr
+                data-test="table-row"
```

Comment:
I am a bit nervous about cases like this where we're adding a bunch of HTML markup for ever row and cell. If we strip these data-test attributes out for production builds, then I'm not worried about it. 

In any case, I _love_ the changes in this PR where the data-test attribute points at a _specific instance_ of something, making the test more traceable/fixable, but I'm wondering if examples like this are adding much value, since the data-test attribute is just a twin of the class. I guess it doesn't hurt, and separates the concerns, so the class (for styling) can be safely changed/renamed/removed/conditional.

I guess it's fine, just a little concerned about the value-to-markup ratio. If we strip out the markup for prod, it's not a problem to me.

Reply:
I think we could tackle it with a babel plugin such as this one: https://www.npmjs.com/package/babel-plugin-jsx-remove-data-test-id

Reply:
This babel plugin is a good next step I think with tests. I will put this on my list.